db/archives.go

```diff
@@ -82,6 +77,12 @@ func (f *ArchiveFilter) Query() string {
 		wheres = append(wheres, fmt.Sprintf("expires_at < $%d", n))
 		n++
 	}
+	limit := ""
+	if f.Limit != "" {
+		limit = fmt.Sprintf(" LIMIT $%d", n)
+		n++
+	}
+
```

Comment:
Can you add an associated test for this, to prevent it from breaking in the future? Probably something using all the filters + limit. Also a comment on the limit block indicating why it must be last